scripts/check-truffle-namespace-dependency-versions.js

```diff
@@ -0,0 +1,136 @@
+const fs = require("fs");
+const path = require("path");
+
+const chalk = require("chalk");
+
+const { Range } = require("semver");
+
+const debug = require("debug")("check-package-versions");
+
+function main() {
+  const packages = readPackages();
+  const errors = [];
+  const warnings = [];
+
+  for (const candidateName in packages) {
+    for (const dependencyName in packages) {
+      try {
+        checkDependencyVersion(
+          packages[candidateName],
+          packages[dependencyName]
+        );
+      } catch (err) {
+        errors.push(err.message);
+      }
+      try {
+        checkDependencyVersionRange(
+          packages[candidateName],
+          packages[dependencyName]
+        );
+      } catch (err) {
+        warnings.push(err.message);
+      }
+    }
+  }
+
+  for (const error of errors) {
+    console.error(chalk.red("Error:"), error);
+  }
+  for (const warning of warnings) {
+    console.error(chalk.yellow("Warning:"), warning);
+  }
+
+  const doneColor =
+    errors.length > 0
+      ? chalk.red
+      : warnings.length > 0
+      ? chalk.yellow
+      : chalk.green;
+  console.error(
+    doneColor("Done."),
+    `${errors.length} errors, ${warnings.length} warnings found for @truffle namespace version dependencies.`
+  );
+
+  if (errors.length > 0) {
+    process.exit(1);
+  }
+}
+
+function getPackageNames() {
```

Comment:
Should this be named something like `getPackageDirectories`? Since we're not actually reaching into `package.json` to find the `name` field here.